### PR TITLE
Item search

### PR DESCRIPTION
**Pull request by HappyNimbus**
@dokdek Please review. Item search should search for the correct item, and output all the details of that item. Also, I cant seem to push it to main, maybe we can delete that and use "add" as the main. Let me know a fix.

### CODE REVIEW

**Comment by dokdek** [comment]
Main is supposed to be branch protected. No one can push to it, you have to create a PR to merge it in.

File: dist/index.js
Comment (dokdek) [inline dist/index.js:1]: Please don't include the /dist folder into github, add this to your gitignore please, these are compiled ts files so they are not needed on git, they can be compiled before use on local machine

File: index.ts [hunk 1]
```diff
@@ -51,6 +52,26 @@ app.post(
     }
 )
 
+app.post(
+    "/searchItemName",
+    async (
+        req: Request<unknown, unknown, itemStuff, unknown>,
```
Comment (dokdek) [inline index.ts]: Can you rename itemStuff to a better name please?

File: searchItemName.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+});
+
+export async function searchItemName(
+    itemName: string,
+
+): Promise<number> {
+
+    return new Promise((resolve,reject) => {
+        db.get(
+            "SELECT *FROM items WHERE itemName=?",
```
Comment (dokdek) [inline searchItemName.ts]: Can we format this string a bit better? Also this SQL query returns exact name matches, usually for searches we want to be a bit lenient and not get only exact matches. Ideall we would do a [fuzzy search](https://www.placekey.io/blog/sql-address-matching), but that would take too much time atm, so we can leave it for later. For now, I suggest using a more lenient matching operator with LIKE %name%. [Reference](https://www.w3schools.com/sql/sql_like.asp)

File: searchItemName.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            [itemName],
+            (error, row: itemDBrow) => {
+                if (error){
+                    console.log(error);
+                    return reject(400);
+                }
+                
+                if(row && row.itemName){
```
Comment (dokdek) [inline searchItemName.ts]: Why is it also checking for row.itemName to exist?

File: searchItemName.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                    console.log("item searched");
+                    console.log(row.itemID)
+                    console.log(row.itemName)
+                    console.log(row.itemPrice)
+                    console.log(row.itemDesc)
+                    return resolve(200);
```
Comment (dokdek) [inline searchItemName.ts]: This doesn't return the item back to the server? You are just returning a status code 200

File: index.ts [hunk 5]
```diff
@@ -51,6 +52,26 @@ app.post(
...
+        res: Response
+    ) => {
+        try{
+            const response = await searchItemName(req.body.itemName);
+            if(response == 200){
+                const token = jwt.sign(req.body.itemName, "memes");
```
Comment (dokdek) [inline index.ts]: Why are you issuing a jwt on item search?

**Review by dokdek — CHANGES_REQUESTED** [review]
@HappyNimbus Please review, thanks

File: index.ts (hunk 1 above)
Comment (HappyNimbus) [reply index.ts]: Done.

File: searchItemName.ts (hunk 3 above)
Comment (HappyNimbus) [reply searchItemName.ts]: I thought it was needed for searching, my mistake, it has been removed.

File: index.ts (hunk 5 above)
Comment (HappyNimbus) [reply index.ts]: Left behind from testing, it has been removed.

File: index.ts (hunk 1 above)
Comment (HappyNimbus) [reply index.ts]: Done, chagned to itemInfo

**Pull request closed without merging**